homeassistant/components/xiaomi_miio/device.py

```diff
@@ -89,3 +92,60 @@ def device_info(self):
             device_info["connections"] = {(dr.CONNECTION_NETWORK_MAC, self._mac)}
 
         return device_info
+
+
+class XiaomiCoordinatedMiioEntity(CoordinatorEntity):
+    """Representation of a base a coordinated Xiaomi Miio Entity."""
+
+    def __init__(self, name, device, entry, unique_id, coordinator):
+        """Initialize the coordinated Xiaomi Miio Device."""
+        super().__init__(coordinator)
+        self._device = device
+        self._model = entry.data[CONF_MODEL]
+        self._mac = entry.data[CONF_MAC]
+        self._device_id = entry.unique_id
+        self._unique_id = unique_id
+        self._name = name
+        self._available = None
+
+    @property
+    def unique_id(self):
+        """Return an unique ID."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the name of this entity, if any."""
+        return self._name
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        device_info = {
+            "identifiers": {(DOMAIN, self._device_id)},
+            "manufacturer": "Xiaomi",
+            "name": self._name,
```

Comment:
```suggestion
            "name": self._device_name,
```

![device_name](https://user-images.githubusercontent.com/478555/122745458-50a0cc80-d289-11eb-9b0b-fc7df7af9e2d.png)